src/Groups/matrices/MatGrp.jl

```diff
@@ -588,10 +588,27 @@ end
 
 """
     general_linear_group(n::Int, q::Int)
-    general_linear_group(n::Int, F::FqNmodFiniteField)
+    general_linear_group(n::Int, F::Ring)
     GL = general_linear_group
 
 Return the general linear group of dimension `n` either over the field `F` or the field `GF(q)`.
+
+Currently, this function only supports fields of type FqNmodFiniteField, however it will eventually support any ring.
```

Comment:
"Any" ring, I am not sure we want to make that promise... I mean, sure, in principle we could create a group object that calls itself `GL(n,R)` for any ring, but its functionality would be extremely limited.
```suggestion
Currently, this function only supports fields of type `FqNmodFiniteField`.
```
Same applies elsewhere